src/app/productConfig/productConfigModal/productConfig.modal.component.js

```diff
@@ -15,53 +14,49 @@ import desigSrcDirective from 'common/directives/desigSrc.directive';
 import showErrors from 'common/filters/showErrors.filter';
 import { giftAddedEvent, cartUpdatedEvent } from 'common/components/nav/navCart/navCart.component';
 
-let controllerName = 'productConfigController';
-export let giveGiftParams = {
+import template from './productConfig.modal.tpl';
+
+const componentName = 'productConfigModal';
+
+export const giveGiftParams = {
   designation: 'd',
   amount:      '$',
   frequency:   'f',
   day:         'dd'
 };
 
-class ModalInstanceCtrl {
+class ProductConfigModalController {
 
   /* @ngInject */
-  constructor( $location, $scope, $log, $uibModalInstance, designationsService, cartService, modalStateService, gettext, productData, nextDrawDate, suggestedAmounts, itemConfig, isEdit, uri ) {
+  constructor( $location, $scope, $log, designationsService, cartService, modalStateService ) {
     this.$location = $location;
     this.$scope = $scope;
     this.$log = $log;
-    this.gettext = gettext;
-    this.$uibModalInstance = $uibModalInstance;
     this.designationsService = designationsService;
     this.cartService = cartService;
     this.modalStateService = modalStateService;
     this.possibleTransactionDays = possibleTransactionDays;
     this.startDate = startDate;
 
-    this.productData = productData;
-    this.nextDrawDate = nextDrawDate;
-    this.itemConfig = itemConfig;
-    this.isEdit = isEdit;
-    this.uri = uri;
-    this.suggestedAmounts = suggestedAmounts;
     this.selectableAmounts = [50, 100, 250, 500, 1000, 5000];
   }
 
-  $onInit() {
-    if ( this.isEdit ) {
-      this.submitLabel = this.gettext( 'Update Gift' );
-    } else {
-      this.submitLabel = this.gettext( 'Add to Gift Cart' );
-      this.initializeParams();
-    }
+  $onInit(){
+    this.productData = this.resolve.productData;
+    this.itemConfig = this.resolve.itemConfig;
+    this.isEdit = this.resolve.isEdit;
+    this.uri =  this.resolve.uri;
+    this.suggestedAmounts =  this.resolve.suggestedAmounts;
```

Comment:
Does this still resolve the dependencies before instantiating the component/controller?